tests/unit/utils/test_urlutils.py

```diff
@@ -787,6 +787,20 @@ def test_hypothesis(self, source):
             assert parsed == source
 
 
+@pytest.mark.parametrize('url, is_pretty, expected', [
```

Comment:
other test cases you could add:
* good and params present
* `?` with no params after it
* param with key but no value?
* mailto with params